reco_utils/evaluation/spark_diversity_evaluation.py

```diff
@@ -353,6 +365,9 @@ def catalog_coverage(self):
     def distributional_coverage(self):
         """Calculate distributional coverage for recommendations across all users.
 
+        The metric definition is based on formula (21) in the following reference:
+            - G. Shani and A. Gunawardana, Evaluating Recommendation Systems, Recommender Systems Handbook pp. 257-297, 2010.
+
         Returns:
             float: distributional coverage
         """
```

Comment:
Better use 
`        d_coverage = -df_entropy.agg(`
rather than 
`        d_coverage = -1 * df_entropy.agg(`
in line 383 below.

Reply:
Also if `p(i) == 0` the code below will give you `None`. You need something like
```
eps = np.finfo(float).eps
F.col("p(i)") * F.log2(F.col("p(i)") + eps)
```

Reply:
# In reco_df, how  many times each col_item is being recommended
df_itemcnt_reco = self.reco_df.groupBy(self.col_item).count()

"p(i)" = F.col("count") / count_row_reco

F.col("count")  cannot be 0, therefore p(i) will not be 0.